SingularityService/src/main/java/com/hubspot/singularity/data/history/SingularityTaskHistoryPersister.java

```diff
@@ -184,8 +196,10 @@ protected boolean moveToHistory(SingularityTaskId object) {
       LOG.debug("Moving {} to history", object);
       try {
         historyManager.saveTaskHistory(taskHistory.get());
+        persisterSuccess = true;
```

Comment:
This ends up making things the same as before. Due to the fact that we catch and continue the loop, we can end up with a situation like the following:

item 1 succeeds -> persisterSuccess = true
item 2 fails -> persisterSuccess = false
item 3 succeeds -> persisterSuccess = true
end -> mark as success

So in that case we are losing the fact that item 2 failed. Might be able to do it easier with a variable that starts as true and is just turned to false if we ever enter one of the catch blocks.

Side question, why make this a class level variable vs just initializing a boolean on each run?

Reply:
Oh yes, that makes much more sense; I'll update.

I can update to initialize on each run, thinking about it now it's probably better for gc, I was trying to reduce some code copying.